Userland/Libraries/LibWeb/PixelUnits.cpp

```diff
@@ -14,19 +14,33 @@ static constexpr i32 fixed_point_denominator = 1 << fractional_bits;
 
 static constexpr i32 radix_mask = fixed_point_denominator - 1;
 
+static constexpr i32 max_integer_value = NumericLimits<int>::max() >> fractional_bits;
+static constexpr i32 min_integer_value = NumericLimits<int>::min() >> fractional_bits;
+
 CSSPixels::CSSPixels(int value)
 {
-    m_value = value * fixed_point_denominator;
+    if (value > max_integer_value) [[unlikely]]
+        m_value = NumericLimits<int>::max();
+    else if (value < min_integer_value) [[unlikely]]
+        m_value = NumericLimits<int>::min();
+    else
+        m_value = static_cast<int>(value) << fractional_bits;
 }
 
 CSSPixels::CSSPixels(unsigned int value)
 {
-    m_value = value * fixed_point_denominator;
+    if (value > max_integer_value) [[unlikely]]
+        m_value = NumericLimits<int>::max();
+    else
+        m_value = static_cast<int>(value) << fractional_bits;
 }
 
 CSSPixels::CSSPixels(unsigned long value)
 {
-    m_value = value * fixed_point_denominator;
+    if (value > max_integer_value) [[unlikely]]
+        m_value = NumericLimits<int>::max();
+    else
+        m_value = static_cast<int>(value) << fractional_bits;
```

Comment:
seems like `static_cast` is not needed here.

Reply:
Sound weird, but I can remove that

Reply:
![grafik](https://github.com/SerenityOS/serenity/assets/37629766/0a9ce11a-3635-4005-8b29-4f254c6b1f6f)

Reply:
Clang-tidy says no

Reply:
oh, my mistake. thank you for checking!